gensim/models/phrases.py

```diff
@@ -133,19 +133,32 @@ def __init__(self, sentences=None, min_count=5, threshold=10.0,
         `delimiter` is the glue character used to join collocation tokens, and
         should be a byte string (e.g. b'_').
 
+        `recode_to_utf8` is an optional parameter (default True) for any2utf8 conversion of input sentences
+        into bytestrings. You may use `recode_to_utf8=False` for ~1.8-2x speed up.
         """
         if min_count <= 0:
-            raise ValueError("min_count should be at least 1")
+            min_count = 1
+            warnings.warn("min_count should be at least 1. Continuing with min_count=1")
 
         if threshold <= 0:
             raise ValueError("threshold should be positive")
 
+        self.recode_to_utf8 = recode_to_utf8
         self.min_count = min_count
         self.threshold = threshold
         self.max_vocab_size = max_vocab_size
         self.vocab = defaultdict(int)  # mapping between utf8 token => its count
         self.min_reduce = 1  # ignore any tokens with count smaller than this
         self.delimiter = delimiter
+        self.is_bytes = True  # for storing encoding type in vocab for supporting both unicode and bytestring input
```

Comment:
Comment hard to understand, and I think it's because the logic is not clear. What is it that is bytes in `is_bytes`?

Isn't it better to simply create a flag of whether the input sequences are `bytes` or not?

Then the comment becomes `# do the input sentences consist of bytestrings?` which is clear.

Also, this check seems to belong to `learn_vocab`, not here.